generic-factory/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::env::BLOCKCHAIN_INTERFACE;
+use near_sdk::json_types::{Base58CryptoHash, Base58PublicKey, Base64VecU8, U128};
+use near_sdk::serde::{Deserialize, Serialize};
+use near_sdk::{env, near_bindgen, AccountId, Balance, CryptoHash};
+
+#[global_allocator]
+static ALLOC: near_sdk::wee_alloc::WeeAlloc<'_> = near_sdk::wee_alloc::WeeAlloc::INIT;
+
+/// This gas spent on the account creation and contract deployment, the rest goes to the `new` call.
+const GAS_FOR_DEPLOY: u64 = 10_000_000_000_000;
+
+const BLOCKCHAIN_INTERFACE_NOT_SET_ERR: &str = "Blockchain interface not set.";
+
+#[near_bindgen]
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct GenericFactory {}
+
+impl Default for GenericFactory {
+    fn default() -> Self {
+        GenericFactory {}
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(crate = "near_sdk::serde")]
+#[serde(untagged)]
+pub enum AccessKey {
+    FullAccessKey(Base58PublicKey),
+    FunctionCall {
+        public_key: Base58PublicKey,
+        receiver_id: AccountId,
+        method_names: Vec<String>,
+        allowance: U128,
+    },
+}
+
+#[near_bindgen]
+impl GenericFactory {
+    pub fn create(
+        &self,
+        name: AccountId,
+        hash: Base58CryptoHash,
+        access_keys: Vec<AccessKey>,
+        method_name: Option<String>,
+        args: Option<Base64VecU8>,
+    ) {
+        internal_create(
+            &format!("{}.{}", name, env::current_account_id()),
+            &CryptoHash::from(hash),
+            access_keys,
+            method_name,
+            args.map(|args| args.0),
+        );
+    }
+}
+
+/// Stores attached data into blob store and returns hash of it.
+/// Implemented to avoid loading the data into WASM for optimal gas usage.
+#[cfg(target_arch = "wasm32")]
```

Comment:
should we consider exposing some better API for this?

Reply:
we should, but this is outside of this PR

Reply:
https://github.com/near/near-sdk-rs/pull/417 has been sitting for about a month, I can try to push for finalizing scope of the next release and getting there soon

Reply:
actually, you wouldn't need to wait for any release, can just define the interface that the `BLOCKCHAIN_INTERFACE` goes through and it should reduce contract size.

I'll do this now so there is an example, and any workflow to test the functionality would be appreciated

Reply:
https://github.com/near/core-contracts/pull/149 figured PR would be the easiest way to see changes. And the sys interface wasn't needed to be duplicated as it's included in `near_bindgen` right now.